### DISCUSSION TITLE
Aggregate timeseries across multiple documents

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#16744] [General]
Hi,

I have a timeseries query as follows

```
declare timeseries out(u)
{
    from Harvest
    group by '1 months'
    select min(), max(), avg(), count()
}
from index HarvestPlans as p
where p.organisation = 'organisations/1-A'
select out(p)
```

Which returns multiple documents with the Harvest timeseries aggregated as per the query.

Is it possible to aggregate the results of the timeseries aggregations per document on the server, so in effect this query returns a single result. As is the query could return 100's of documents and we're looking to produce 1 chart for all the documents matching the query, trying to avoid aggregating on the client.

I've read about timeseries roll-ups but it looks like this it at the document level too.

Thanks

Nick

**Comment by ayende** [comment]
To aggregate data across documents, you need to use a time series index
See: https://ravendb.net/docs/article-page/5.4/csharp/document-extensions/timeseries/indexing#map-reduce-time-series-index

**Reply by nickchampion** [reply]
Thats what I needed, thank you!
